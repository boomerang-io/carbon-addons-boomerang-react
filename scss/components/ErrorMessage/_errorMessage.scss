@use '@carbon/themes/scss/tokens' as theme;
@use "../../global/config" as *;
@use "../../global/utils";

.#{$prefix}--bmrg-error-message {
  color: theme.$text-primary;
  font-size: utils.rem(24px);
  font-weight: 600;
  line-height: utils.rem(32px);
  margin: 0 10%;
  text-align: center;
}

.#{$prefix}--bmrg-error-message__img {
  height: 8rem;
  width: 8rem;
  fill: theme.$background-brand;
}

.#{$prefix}--bmrg-error-message__text {
  margin: 1rem;
  font-size: utils.rem(24px);
  font-weight: 600;
  line-height: utils.rem(32px);
  text-align: center;
}

.#{$prefix}--bmrg-error-message__status {
  font-size: 1.25rem;
}

.#{$prefix}--bmrg-error-message__subtext {
  font-size: 1rem;
  line-height: utils.rem(20px);
  margin: utils.rem(10px);
}
