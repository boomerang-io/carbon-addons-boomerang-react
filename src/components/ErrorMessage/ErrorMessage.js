import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Warning16 } from '@carbon/icons-react';
import { settings } from 'carbon-components';

const { prefix } = settings;

const Error = ({ status, statusText, className, style, ...rest }) => {
  let message;
  if (status && statusText) {
    message = (
      <div
        className={`${prefix}--bmrg-error-message__status`}
      >{`Status: ${status} ${statusText}`}</div>
    );
  } else if (status) {
    message = <div className={`${prefix}--bmrg-error-message__status`}>{`Status: ${status}`}</div>;
  } else if (statusText) {
    message = <div className={`${prefix}--bmrg-error-message__status`}>{`${statusText}`}</div>;
  }

  const classNames = classnames(`${prefix}--bmrg-error-message`, className);

  return (
    <div className={classNames} style={style} {...rest}>
      <div className={`${prefix}--bmrg-error-message`}>
        <Warning16 className={`${prefix}--bmrg-error-message__img`} alt="Warning" />
        <h2 className={`${prefix}--bmrg-error-message__text`}>Oops, something went wrong.</h2>
        {message}
        <p className={`${prefix}--bmrg-error-message__subtext`}>
          Try reloading the page. And if you could be so kind, please send us an issue report.
        </p>
      </div>
    </div>
  );
};

Error.defaultProps = {
  className: '',
};

Error.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  status: PropTypes.string,
  statusText: PropTypes.string,
};

export default Error;
