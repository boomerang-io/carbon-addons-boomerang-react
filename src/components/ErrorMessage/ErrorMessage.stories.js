import React from 'react';
import { storiesOf } from '@storybook/react';

import ErrorComponent from './index';

storiesOf('ErrorMessage', module).add('default', () => {
  return <ErrorComponent />;
});
