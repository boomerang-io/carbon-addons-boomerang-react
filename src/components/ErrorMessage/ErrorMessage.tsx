import React from "react";
import cx from "classnames";
import { Warning } from "@carbon/react/icons";
import { prefix } from "../../internal/settings";

Error.defaultProps = {
  className: "",
};

type OwnProps = {
  className?: string;
  style?: any;
  status?: string;
  statusText?: string;
};

type Props = OwnProps & typeof Error.defaultProps;

function Error({ status, statusText, className, style, ...rest }: Props) {
  let message;
  if (status && statusText) {
    message = <div className={`${prefix}--bmrg-error-message__status`}>{`Status: ${status} ${statusText}`}</div>;
  } else if (status) {
    message = <div className={`${prefix}--bmrg-error-message__status`}>{`Status: ${status}`}</div>;
  } else if (statusText) {
    message = <div className={`${prefix}--bmrg-error-message__status`}>{`${statusText}`}</div>;
  }

  const classNames = cx(`${prefix}--bmrg-error-message`, className);

  return (
    <div className={classNames} style={style} {...rest}>
      <Warning size={16} className={`${prefix}--bmrg-error-message__img`} alt="Warning" />
      <h2 className={`${prefix}--bmrg-error-message__text`}>Oops, something went wrong.</h2>
      {message}
      <p className={`${prefix}--bmrg-error-message__subtext`}>
        Try reloading the page. And if you could be so kind, please send us an issue report.
      </p>
    </div>
  );
};

export default Error;
