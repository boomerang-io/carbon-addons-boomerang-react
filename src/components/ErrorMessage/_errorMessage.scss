@include exports('addons-bmrg-error') {
  .#{$prefix}--bmrg-error-message {
    color: $text-02;
    font-size: rem(24px);
    font-weight: 600;
    line-height: rem(32px);
    margin: 0 10%;
    text-align: center;
  }

  .#{$prefix}--bmrg-error-message__img {
    height: 8rem;
    width: 8rem;
    fill: $interactive-01;
  }

  .#{$prefix}--bmrg-error-message__text {
    margin: 1rem;
    font-size: rem(24px);
    font-weight: 600;
    line-height: rem(32px);
    text-align: center;
  }

  .#{$prefix}--bmrg-error-message__status {
    font-size: 1.25rem;
  }

  .#{$prefix}--bmrg-error-message__subtext {
    font-size: 1rem;
    line-height: rem(20px);
    margin: rem(10px);
  }
}
