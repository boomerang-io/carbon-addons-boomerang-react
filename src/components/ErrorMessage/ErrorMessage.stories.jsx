import ErrorMessage from "./ErrorMessage";

export default {
  title: "Errors/ErrorMessage",
  component: ErrorMessage,
};

export const Default = (args) => {
  return <ErrorMessage {...args} />;
};
