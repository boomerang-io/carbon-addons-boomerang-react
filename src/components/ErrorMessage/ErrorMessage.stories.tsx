import React from "react";
import ErrorMessage from "./ErrorMessage";

export default {
  title: "Errors/ErrorMessage",
  component: ErrorMessage,
  parameters: {
    docs: {
      description: {
        component: "Error message for errors that are _not_ full page",
      },
    },
  },
};

export const Default = (args: any) => {
  return <ErrorMessage {...args} />;
};
