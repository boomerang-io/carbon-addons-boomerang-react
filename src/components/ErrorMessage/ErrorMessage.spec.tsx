/*
IBM Confidential
694970X, 69497O0
Â© Copyright IBM Corp. 2022, 2024
*/


import React from "react";
import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import ErrorMessage from "./ErrorMessage";

const status = "testStatus";
const statusText = "testText";

describe("ErrorMessage", () => {
  test("snapshot", async () => {
    const { baseElement } = render(<ErrorMessage status={status} statusText={statusText} />);
    expect(baseElement).toMatchSnapshot();
  });
  
  test("functional", async () => {
    const { findByText } = render(<ErrorMessage status={status} statusText={statusText} />);
    const testStatus = await findByText(/testStatus/i);
    expect(testStatus).toBeInTheDocument();
    const testStatusText = await findByText(/testText/i);
    expect(testStatusText).toBeInTheDocument();
  });

  test("a11y", async () => {
    const { container } = render(<ErrorMessage status={status} statusText={statusText} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
