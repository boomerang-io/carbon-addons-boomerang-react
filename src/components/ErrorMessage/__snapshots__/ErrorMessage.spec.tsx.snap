// Vitest Snapshot v1

exports[`ErrorMessage > snapshot 1`] = `
<body>
  <div>
    <div
      class="cds--bmrg-error-message"
    >
      <svg
        alt="Warning"
        aria-hidden="true"
        class="cds--bmrg-error-message__img"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z M8,14c-3.3,0-6-2.7-6-6s2.7-6,6-6s6,2.7,6,6S11.3,14,8,14z"
        />
        <path
          d="M7.5 4H8.5V9H7.5zM8 10.2c-.4 0-.8.3-.8.8s.3.8.8.8c.4 0 .8-.3.8-.8S8.4 10.2 8 10.2z"
        />
      </svg>
      <h2
        class="cds--bmrg-error-message__text"
      >
        Oops, something went wrong.
      </h2>
      <div
        class="cds--bmrg-error-message__status"
      >
        Status: testStatus testText
      </div>
      <p
        class="cds--bmrg-error-message__subtext"
      >
        Try reloading the page. And if you could be so kind, please send us an issue report.
      </p>
    </div>
  </div>
</body>
`;
